host/class/uvc/usb_host_uvc/uvc_descriptor_parsing.c

```diff
@@ -375,3 +402,110 @@ esp_err_t uvc_desc_get_streaming_interface_num(
     }
     return ret;
 }
+
+bool uvc_desc_is_uvc_device(const usb_config_desc_t *cfg_desc)
+{
+    assert(cfg_desc);
+    int offset = 0;
+    int total_len = cfg_desc->wTotalLength;
+
+    const usb_standard_desc_t *current_desc = (const usb_standard_desc_t *)cfg_desc;
+    while ((current_desc = usb_parse_next_descriptor_of_type((const usb_standard_desc_t *)current_desc, total_len, USB_B_DESCRIPTOR_TYPE_INTERFACE, &offset))) {
+        const usb_intf_desc_t *iface_desc = (const usb_intf_desc_t *)current_desc;
+        if (USB_CLASS_VIDEO == iface_desc->bInterfaceClass) {
+            return true;
+        }
+    }
+    return false;
+}
+
+esp_err_t uvc_desc_get_frame_list(const usb_config_desc_t *config_desc, uint8_t uvc_index, uvc_host_frame_info_t (*frame_info_list)[], size_t *list_size)
+{
+    esp_err_t ret = ESP_OK;
+    UVC_CHECK(config_desc && list_size, ESP_ERR_INVALID_ARG);
+    size_t num_frame = 0;
+    size_t frame_index = 0;
+
+    const uvc_vs_input_header_desc_t *input_header = uvc_desc_get_streaming_interface_input_header(config_desc, uvc_index);
+    UVC_CHECK(input_header, ESP_ERR_NOT_FOUND);
+
+    // Find requested Format descriptors
+    int format_offset = 0;
+    const usb_standard_desc_t *current_desc = (const usb_standard_desc_t *)input_header;
+    while ((current_desc = usb_parse_next_descriptor_of_type(current_desc, input_header->wTotalLength, UVC_CS_INTERFACE, &format_offset))) {
+        if (!uvc_desc_is_format_desc(current_desc)) {
+            continue;
+        }
+
+        const uvc_format_desc_t *this_format = (const uvc_format_desc_t *)(current_desc);
+        enum uvc_host_stream_format format_type = uvc_desc_parse_format(this_format);
+        if (UVC_VS_FORMAT_UNDEFINED == format_type) {
+            continue;
+        }
+
+        num_frame = this_format->bNumFrameDescriptors;
+
+        /*!< Just return the list size */
+        if (!frame_info_list) {
+            *list_size = num_frame;
+            return ESP_OK;
+        }
+
+        if (*list_size < num_frame) {
+            ESP_LOGE("uvc_desc", "list_size(%d) < num_frame(%d)", *list_size, num_frame);
+            return ESP_FAIL;
+        }
+
+        // We found required Format Descriptor
+        // Now we look for correct Frame Descriptors which should be directly after Format
+        while ((current_desc = usb_parse_next_descriptor_of_type(current_desc, input_header->wTotalLength, UVC_CS_INTERFACE, &format_offset))) {
+            if (!uvc_desc_is_frame_desc(current_desc)) {
+                break;
+            }
+            uvc_frame_desc_t *this_frame = (uvc_frame_desc_t *)current_desc;
+
+            if (frame_index > *list_size - 1) {
+                return ESP_ERR_NO_MEM;
```

Comment:
Here we should just `break;`

Reply:
If the size of the passed `uvc_host_frame_info_list` is smaller than the sum of all `VS->bNumFrameDescriptors`, should the `list_size` we return be the sum of all `VS->bNumFrameDescriptors` or the actual size of the `frame_lifo_list` that was filled?